src/Parser/Mage/Frost/Modules/Features/BrainFreeze.js

```diff
@@ -11,81 +11,123 @@ const debug = false;
+      this.overwrittenProcs += 1;
+      debug && console.log("Brain Freeze proc overwritten w/ GS with too few icicles @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+    } else {
+      this.okOverwrittenProcs += 1;
+      debug && console.log("Acceptable Brain Freeze proc overwritten w/ GS @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+    }
   }
 
-	on_byPlayer_cast(event) {
-		const spellId = event.ability.guid;
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
     if (spellId !== SPELLS.FLURRY.id) {
-			return;
-		}
-		this.lastFlurryTimestamp = this.owner.currentTimestamp;
-		if (!this.selectedCombatant.hasBuff(SPELLS.BRAIN_FREEZE.id) && !this.selectedCombatant.hasBuff(SPELLS.WINTERS_REACH_BUFF.id)) {
-			this.flurryWithoutProc += 1;
-		}
-	}
-
-	on_byPlayer_removebuff(event) {
-		const spellId = event.ability.guid;
+      return;
+    }
+    this.lastFlurryTimestamp = this.owner.currentTimestamp;
+    if (!this.selectedCombatant.hasBuff(SPELLS.BRAIN_FREEZE.id) && !this.selectedCombatant.hasBuff(SPELLS.WINTERS_REACH_BUFF.id)) {
+      this.flurryWithoutProc += 1;
+    }
+  }
+
+  on_byPlayer_removebuff(event) {
+    const spellId = event.ability.guid;
+    if (spellId === SPELLS.ICICLES_BUFF.id) {
+      this.handleIcicleStacks(event);
+      return;
+    }
     if (spellId !== SPELLS.BRAIN_FREEZE.id) {
-			return;
-		}
+      return;
+    }
     if (!this.lastFlurryTimestamp || this.lastFlurryTimestamp + PROC_WINDOW_MS < this.owner.currentTimestamp) {
-			this.expiredProcs += 1; // it looks like Brain Freeze is always removed after the cast, and always on same timestamp
-      if (debug) {
-        console.log("Brain Freeze proc expired @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
-			}
-		}
+      this.expiredProcs += 1; // it looks like Brain Freeze is always removed after the cast, and always on same timestamp
+      debug && console.log("Brain Freeze proc expired @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+    }
   }
 
-	get overwrittenPercent() {
-		return (this.overwrittenProcs / this.totalProcs) || 0;
-	}
+  get overwrittenPercent() {
+    return (this.overwrittenProcs / this.totalProcs) || 0;
+  }
 
-	get expiredPercent() {
-		return (this.expiredProcs / this.totalProcs) || 0;
-	}
+  get expiredPercent() {
+    return (this.expiredProcs / this.totalProcs) || 0;
+  }
 
-	get wastedProcs() {
-		return this.overwrittenProcs + this.expiredProcs;
-	}
+  get wastedProcs() {
+    return this.overwrittenProcs + this.expiredProcs;
+  }
 
-	get wastedPercent() {
-		return (this.wastedProcs / this.totalProcs) || 0;
-	}
+  get wastedPercent() {
+    return (this.wastedProcs / this.totalProcs) || 0;
+  }
 
-	get usedProcs() {
-		return this.totalProcs - this.wastedProcs;
-	}
+  get usedProcs() {
+    return this.totalProcs - this.wastedProcs - this.okOverwrittenProcs;
```

Comment:
I might not be thinking straight because im sick .... but is this accurate??

If im reading this right, if you have GS and you refresh the buff with 1-2 stacks then you are marking it as an OverwrittenProc and if you have 3,4,5 then its an OKOverwrittenProc, therefore it wouldnt make sense to subtract the OKOverwrittenProcs from the totalprocs because thats going to lower the number of properly used procs.

So shouldnt usedProcs be totalProcs - wastedProcs?

Also, i dont think you need to count the number of OKOverwrittenProcs unless you are planning on listing that number somewhere. If not, it should be enough to just not count them as an overwrittenProc and therefore not have it count against them.

Reply:
I am tracking OKOverwrittenProcs and OverwrittenProcs separately, but an OKOverwrittenProc is not a used proc which is why it gets subtracted. And yes I do display it somewhere, it shows up like this: 
![image](https://user-images.githubusercontent.com/43309639/45600435-79e1ef00-b9b1-11e8-9964-3f8ac7a94fe1.png)